include/spdlog/details/os-inl.h

```diff
@@ -484,15 +484,15 @@ SPDLOG_INLINE void utf8_to_wstrbuf(string_view_t str, memory_buf_t &target)
     }
 
     int result_size = static_cast<int>(target.capacity());
-    if ((str_size + 1) * 2 < result_size)
+    if (str_size + 1 > result_size)
```

Comment:
Not sure what will happen in str_size is exatcly MAX_INT. 

To be on the safe side, , I think the 

```c++
if (str.size() > static_cast<size_t>((std::numeric_limits<int>::max)()))
    {
        throw_spdlog_ex("UTF-8 string is too big to be converted to UTF-16");
    }
```
should be changed to something like (don't allow sizes bigger than MAX_INT/2):


```c++
if (str.size() > static_cast<size_t>((std::numeric_limits<int>::max/2)()))
    {
        throw_spdlog_ex("UTF-8 string is too big to be converted to UTF-16");
    }
```

Reply:
Why not add a -1, like this?
```cpp
if (str.size() > static_cast<size_t>(std::numeric_limits<int>::max() - 1))
```

Reply:
agreed

Reply:
Resolved